indico/modules/events/contributions/controllers.py

```diff
@@ -175,3 +182,61 @@ class RHContributionPersonList(RHContributionPersonListMixin, RHManageContributi
     def _membership_filter(self):
         contribution_ids = {contrib.id for contrib in self.contribs}
         return Contribution.id.in_(contribution_ids)
+
+
+class RHContributionProtection(RHManageContributionBase):
+    """Manage contribution protection"""
+
+    def _process(self):
+        form = ContributionProtectionForm(obj=FormDefaults(**self._get_defaults()))
+        if form.validate_on_submit():
+            self._handle_contrib_managers(form.managers.data)
+            self._handle_protection_mode(form.protection_mode.data)
+            self._handle_access_control_list(form.acl.data)
+            self._handle_contrib_submitters(form.submitters.data)
+            return jsonify_data(flash=False)
+        protection_message = _render_contrib_protection_message(self.contrib)
+        return jsonify_template('events/contributions/management/contribution_protection.html', form=form,
+                                protection_message=protection_message)
+
+    def _get_defaults(self):
+        managers = {p.principal for p in self.contrib.acl_entries if p.full_access}
+        submitters = {p.principal for p in self.contrib.acl_entries if 'submit' in p.roles}
+        acl = {p.principal for p in self.contrib.acl_entries if p.read_access}
+        return {'managers': managers, 'submitters': submitters, 'protection_mode': self.contrib.protection_mode.name,
+                'acl': acl}
+
+    def _handle_contrib_managers(self, managers_data):
+        current_managers = {(acl.principal.name if acl.principal.is_group else acl.principal.id): acl.principal
+                            for acl in self.contrib.acl_entries if acl.full_access}
+        new_managers = {(m.group.name if m.is_group else m.id): m for m in managers_data}
+        deleted_managers_ids = current_managers.viewkeys() - new_managers.viewkeys()
+        for deleted_manager_id in deleted_managers_ids:
+            self.contrib.update_principal(current_managers[deleted_manager_id], full_access=False)
+        for new_manager_id in new_managers.viewkeys() - current_managers.viewkeys():
+            self.contrib.update_principal(new_managers[new_manager_id], full_access=True)
+
+    def _handle_protection_mode(self, protection_mode):
+        if ProtectionMode.get(protection_mode) is not None:
```

Comment:
also, calling .get() twice with the same arguments is rarely a good idea (there's this thing called "variables" ;))